java/vector/src/main/java/org/apache/arrow/vector/VectorSchemaRoot.java

```diff
@@ -47,6 +47,17 @@ public VectorSchemaRoot(FieldVector parent) {
     }
   }
 
+  public VectorSchemaRoot(List<Field> fields, List<FieldVector> fieldVectors) {
```

Comment:
added static method to create from schema. Left this constructor + constructor from parent field since lots of tests were using that.